text/3463-crates-io-policy-update.md

```diff
@@ -0,0 +1,256 @@
+- Start Date: 2023-07-24
+- RFC PR: [rust-lang/rfcs#3463](https://github.com/rust-lang/rfcs/pull/3463)
+
+# Summary
+[summary]: #summary
+
+The Rust community has outgrown the current crates.io policies. This RFC proposes a new "Terms of Use" policy based on prior work by PyPI, npm and GitHub.
+
+# Motivation
+[motivation]: #motivation
+
+> Why are we doing this? What use cases does it support? What is the expected outcome?
+
+crates.io has a "[Package Policies](https://crates.io/policies)" page, which describes the current, organically grown policies. A lot of support requests or questionable uses of crates.io however need explicit decisions from the crates.io team, since many cases are currently not covered by these policies. Additionally, decisions made by the team may be seen as arbitrary without written guidelines.
+
+The situation around name squatting has lately also reached unsustainable levels, and while [namespaces](https://github.com/rust-lang/rfcs/pull/3243) might help with some parts of the problem we would still need policies for name squatting namespaces then.
+
+The main motivation for this RFC is to give the crates.io team a fixed set of rules to determine if a project is using crates.io in a reasonable way, or if the user should get a warning and the project potentially be removed. It is mostly codifying the existing practices of the team, except for being more strict regarding name squatting.
+
+# Proposal
+[guide-level-explanation]: #guide-level-explanation
+
+The following is a proposed new "Terms of Use" policy for crates.io, replacing <https://crates.io/policies> and <https://crates.io/data-access>.
+
+-------------------------------------------------------------------------------
+
+# Terms of Use
+
+**Short version:** _crates.io is a critical resource for the Rust ecosystem,
+which hosts a variety of packages from a diverse group of users. That resource
+is only effective when our users are able to work together as part of a
+community in good faith. While using crates.io, you must comply with our
+Acceptable Use Policies, which include some restrictions on content and conduct
+on crates.io related to user safety, intellectual property, privacy,
+authenticity, and other limitations. In short, be excellent to each other!_
+
+We do not allow content or activity on crates.io that:
+
+- violates the [Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct)
+  of the Rust project
+- is unlawful or promotes unlawful activities, incurring legal liability in the
```

Comment:
Shouldn't those countries be explicitly spelled out to give people a chance to actually comply? Most countries do have some rather weird laws and it is not super obvious which "countries the Rust Foundation officially operates in". Also this could also be subject to change at any time.

Reply:
>Shouldn't those countries be explicitly spelled out to give people a chance to actually comply?

That makes sense to me. (The list is the US and the UK, for the record.)

Reply:
After talking with the foundation about this, it has become clear that this is a question that might not be easy to answer. The Rust Foundation is indeed a legal entity in the US and UK, but they do employ people in other countries as well and that could be seen as "operating" in these countries. 

The original proposal only used "is unlawful or promotes unlawful activities" (like https://pypi.org/policy/acceptable-use-policy/), but some people suggested that the suffix with the foundation would be useful.

Ultimately, in questions on legality the crates.io team will have to involve the legal team of the Rust Foundation to figure out if something is unlawful or not, but when in doubt we won't simply delete a crate and instead contact the author and try to come to an agreement of some sort.